### DISCUSSION TITLE
Error code: 401 - {'error': {'message': 'Missing Authentication header or invalid API key', 'code': 401}}

### DISCUSSION THREAD

**Discussion by exponentiallyio** [danielmiessler/Fabric#656] [Q&A]
Hello, it was working fine, then I used:
`fabric --changeDefaultModel claude-3.5-sonnet`

Now it throws this error whenever I run it:
`Error: Error code: 401 - {'error': {'message': 'Missing Authentication header or invalid API key', 'code': 401}}
Error code: 401 - {'error': {'message': 'Missing Authentication header or invalid API key', 'code': 401}}`

I've tried running the setup again, changing the Default Model back, still the same error. And I've checked the API keys in ~/.config/fabric/.env and they're there

Would appreciate any help!

**Comment by exponentiallyio** [comment]
Hello, does anyone have any insight/suggestions on this? Thank you

**Comment by patonw** [comment]
Do you have an OpenAI key setup also or just Anthropic? It looks like fabric checks to see if OpenAI offers that model, and doesn't handle a missing API key: https://github.com/danielmiessler/fabric/blob/dd4b896f4dfa2923272743b7fa4f92a2db4aef86/installer/client/cli/utils.py#L415-L430

Also, claude models need to be suffixed by the model build number. In this case you'd need `claude-3-5-sonnet-20240620`.

**Reply by exponentiallyio** [reply]
Thanks @patonw . I've set up all the API keys. And I've tried openai, llama, claude models, with the same issue

It was all working fine until I changed the default model and now none of them work, puzzling

Also - I noticed I don't have claude-3-5-sonnet-20240620 when I use --listmodels, do I need to update it somehow?

Thanks!

**Reply by patonw** [reply]
Ah, if `--listmodels` doesn't include the following, then something is wrong with the configuration or configured key:
```
Claude Models:
claude-3-5-sonnet-20240620
claude-3-opus-20240229
claude-3-sonnet-20240229
claude-3-haiku-20240307
claude-2.1
```

The code simply checks for the existence of the key: 
https://github.com/danielmiessler/fabric/blob/dd4b896f4dfa2923272743b7fa4f92a2db4aef86/installer/client/cli/utils.py#L409-L413

Maybe dotenv is not able to find the file or the file is malformed. Check that any quote marks are closed, etc.
